src/Logic.hs

```diff
@@ -380,9 +380,10 @@ handlePullRequestClosed closingReason pr state = do
   -- actually delete the pull request
   pure . Pr.deletePullRequest pr
        $ case Pr.lookupPullRequest pr state of
-         Just (Pr.PullRequest{Pr.integrationStatus = Promoted}) -> state
-         -- we unintegrate PRs after it if it has not been promoted to master
-         _ -> unintegrateAfter pr $ state
```

Comment:
What would "after" even mean if a PR wasn't integrated?

Reply:
Hoff's internal terminology is a bit confusing:

* integrated mean: a `testing/123` branch has been created with a rebase and merge commit
* promoted means: the merge commit sha has been pushed to be the new master

So integrated after it means the further testing branches, e.g.:

* `testing/1` is integrated on top of `master`
* `testing/2` is integrated on top of `testing/1`
* `testing/3` is integrated on top of `testing/2`

What we want to do here with `unintegrateAfter` is, for example, supposing PR\#2 is closed.  To "unintegrate" `testing/3` so it is scheduled to be integrated on top of `testing/1` (assuming the build of PR\#1 has not been concluded)

(But sometimes, sometimes, integrated is used to mean promoted.  I should probably hunt these cases and make the terminology more consistent later.)

Reply:
> What would "after" even mean if a PR wasn't integrated?

Yes, exactly!  So we end up restarting the builds "randomly" because an unintegrated PR received a new push.